sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -53,27 +49,32 @@
     private final ClientLogger logger = new ClientLogger(TableAsyncClient.class);
     private final String tableName;
     private final AzureTableImpl implementation;
-    private final TablesImpl tableImplementation;
     private final String accountName;
     private final String tableUrl;
-    private final TablesServiceVersion apiVersion;
     private final QueryOptions defaultQueryOptions = new QueryOptions()
         .setFormat(OdataMetadataFormat.APPLICATION_JSON_ODATA_FULLMETADATA);
 
     TableAsyncClient(String tableName, AzureTableImpl implementation) {
+        try {
+            final URI uri = URI.create(implementation.getUrl());
+            this.accountName = uri.getHost().split("\\.", 2)[0];
+            this.tableUrl = uri.resolve("/" + tableName).toString();
+            logger.verbose("Table Service URI: {}", uri);
+        } catch (IllegalArgumentException ex) {
+            throw logger.logExceptionAsError(ex);
+        }
+
         this.implementation = implementation;
-        this.tableImplementation = implementation.getTables();
         this.tableName = tableName;
-        this.accountName = null;
-        this.tableUrl = null;
-        this.apiVersion = null;
     }
 
     TableAsyncClient(String tableName, HttpPipeline pipeline, String url, TablesServiceVersion serviceVersion,
         SerializerAdapter serializerAdapter) {
 
         try {
             final URI uri = URI.create(url);
+            this.accountName = uri.getHost().split("\\.", 2)[0];
```

Comment:
nit: Change this to a constant "\\." so that it can be reused in both ctors locations.

Reply:
Resolved in PR #14497